src/engine/filters/enginefiltermoogladder4.h

```diff
@@ -169,12 +169,12 @@ class EngineFilterMoogLadderBase : public EngineObjectConstIn {
                 double new2 = processSample(pIn[i+1], &m_buf[1]);
 
                 if (i < iBufferSize / 2) {
-                    pOutput[i] = old1;
-                    pOutput[i + 1] = old2;
+                    pOutput[i] = static_cast<CSAMPLE>(old1);
```

Comment:
should old1 here, and similar temp values, all be CSAMPLE and prevent the need for casting at all?

Reply:
I thought about that as well, but I was unsure if this leads to additional rounding errors that pile up. No idea if this is neglible or not in this case.

Reply:
When I remember right, the filter need double precision internally to sound good. 
But old1 always stored float so it is reasonable to use float here.